bin/scanpy-scale-data.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import logging
+import scanpy.api as sc
+from scanpy_wrapper_utils import ScanpyArgParser, comma_separated_list
+from scanpy_wrapper_utils import read_input_object, write_output_object
+
+
+def main(args):
+    logging.debug(args)
+
+    adata = read_input_object(args.input_object_file, args.input_format)
+
+    if args.do_log:
+        sc.pp.log1p(adata)
+
+    if args.var_to_regress:
+        sc.pp.regress_out(adata, args.var_to_regress)
+
+    sc.pp.scale(adata, zero_center=args.zero_center, max_value=args.scale_max)
```

Comment:
Question: are we sure that we don't want to wrap regress_out() and scale() independently of one another? Are they universally used alongside each other and never independently? What if I want to regress but not scale?

Reply:
It's usually imperative to scale the data before PCA to ensure top PCs are not just dominated by genes with large nominal variance, while regressing out certain variable is optional and problem-dependent, hence the design choice.

Reply:
Hi @nh3 . The thing is, that's you making a methodological choice that's locking future workflow builders into your assertions (even if they're sensible ones). Personally, I would rather people could build workflows in a similar manner as is possible from within Scanpy itself. If you look at the R-based workflows we've mostly kept to a single wrapper per function, with the wrapper named for that function, and I would really think we should apply the same philosophy here- unless the number of wrappers required would be prohibitively large. 

@pcm32 - what do you think?

Reply:
I think that there is a balance between granularity and simplicity, not always easy to strike I guess. On one side one could provide separate functions for scale and regress IFF they are going to be used for other purpose separately. The case to the extreme of granularity is being implemented within the scanpy Galaxy wrappers by Bjorn and Berenice, where each scanpy function is wrapped separately. While this gives the user extreme flexibility, in my view it also makes operation more complicated as the user needs to know all the details of Scanpy (instead of just the main core operations that he/she wants to enact). If scale is the only method that will use regress, might not be worth it to add the extra boilerplate. Remember that the more boxes that you put on a workflow, the less readable and clear it becomes in terms of its operational flow. Looking at Seurat scripts, I see that there is no separate regress and that is indeed only included in the scale wrapper, so I would be happy to proceed likewise in this case. We can always add it later as a separate method if needed, but if we don't have a use case, I wouldn't bother with the extra granularity. But then I'm not an expert in the usage of these tools.

Reply:
The flexible approach is what appeals to me- but I'll go with the group view ;-)